realtime.js

```diff
@@ -91,27 +91,35 @@ exports.help = function(entry_id, ta) {
     });
 };
 
-exports.fixq = function(entry_id) {
+exports.request_update = function(entry_id) {
     exports.seq = exports.seq + 1;
     if(!sio) {
         console.log("ERROR: Socket.io is not initialized yet");
         return;
     }
-    sio.emit("fixq", {
+    sio.emit("request-update", {
         seq: exports.seq,
         id: entry_id
     });
 };
 
-exports.update = function(entry_id) {
+exports.update = function(entry_id, updated_question) {
     exports.seq = exports.seq + 1;
     if(!sio) {
         console.log("ERROR: Socket.io is not initialized yet");
         return;
     }
-    sio.emit("update-question", {
+    
+    // Students: only need to update seq
+    sio.to(student_room).emit("update-question-student", {
+        seq: exports.seq
+    });
+
+    // TAs: need to pass the updated question
+    sio.to(ta_room).emit("update-question-ta", {
```

Comment:
You don't need to use two separate message names for this, since we can just check if the user is a TA on the client.